R/simulated_annealing.R

```diff
@@ -0,0 +1,76 @@
+#' Feature selector that performs simmulated annealing to select features.
+#' 
+#' @description Algorithm randomly chooses a set of features, trains on them, scores the model.
+#'Then the algorithm slightly modifies the chosen features randomly and tests to see
+#'if the model improves. If there is improvement, the newer model is kept, if not the 
+#'algorithm tests to see if the worse model is still kept based on a acceptance 
+#'probability that decreases as iterations continue and if the model performs worse.
+#' 
+#' @param scorer  A custom user-supplied function that accepts X and y (as defined below) as input and returns the error of the datasets.
+#' @param X data frame of training features
+#' @param y data frame of training targets
+#' @param c control rate of feature perturbation
+#' @param iterations number of iterations
+#' @param bools If true function returns array of boolean values instead of column indicies
+#' 
+#' @return array of selected features
+#' @export
+#' 
+#' @examples
+#' 
+simulated_annealing <- function(scorer, X, y, c = 1, iterations = 100, bools = FALSE) {
+    
+    # Is `scorer` a function?
+    if (class(scorer) != "function") {
+    stop("Expected a function for `scorer`." )
+    }
+
+    # Do we have a data.frame or something compatible like tibble?
+    if (!any(class(X) == "data.frame")) {
+        stop("Expected a `data.frame` object for `data`.")
+    }
+
+    # Set mutate percentage
+    mutate <- 0.05
```

Comment:
Just wandering how this percentage was selected.

Reply:
Typically between 0.01 - 0.05.
(http://www.feat.engineering/simulated-annealing.html)
0.05 chosen arbitrarily for now.

Reply:
thanks for sharing!